hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1866,6 +1880,112 @@ int BleObject::ConnectionsManager::getConnectionInfo(hal_ble_conn_handle_t connH
     return SYSTEM_ERROR_NONE;
 }
 
+int BleObject::ConnectionsManager::setPairingConfig(const hal_ble_pairing_config_t* config) {
+    CHECK_TRUE(config, SYSTEM_ERROR_INVALID_ARGUMENT);
+    ioCaps_ = config->io_caps;
+    return SYSTEM_ERROR_NONE;
+}
+
+int BleObject::ConnectionsManager::startPairing(hal_ble_conn_handle_t connHandle) {
+    BleConnection* connection = fetchConnection(connHandle);
+    CHECK_TRUE(connection, SYSTEM_ERROR_NOT_FOUND);
+    CHECK_TRUE(connection->pairState == BLE_PAIRING_STATE_NOT_INITIATED ||
+               connection->pairState == BLE_PAIRING_STATE_REJECTED ||
+               connection->pairState == BLE_PAIRING_STATE_SEC_REQ_RECEIVED, SYSTEM_ERROR_INVALID_STATE);
+    ble_gap_sec_params_t secParams = {};
+    LOG_DEBUG(TRACE, "Own security params: %d, %d, %d, %d, %d, %d", secParams.bond, secParams.mitm, secParams.lesc, secParams.io_caps, secParams.keypress, secParams.oob);
```

Comment:
This debug always outputs all zero, since it's right after creation. It could be placed a few lines down (right before the call to `sd_ble_gap_authenticate()` to Log the actual parameters.